synapse/storage/databases/main/event_federation.py

```diff
@@ -1292,6 +1292,79 @@ def _get_backfill_events(
 
         return event_id_results
 
+    @trace
+    async def record_event_backfill_attempt(self, event_id: str) -> None:
```

Comment:
docstring please, but:
```suggestion
    async def record_event_failed_backfill_attempt(self, room_id: str, event_id: str) -> None:
```